core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -982,6 +982,7 @@ private Object readResolve() {
         if (jdks == null) {
             jdks = new ArrayList<>();
         }
+        slaveAgentPort = SystemProperties.getInteger(Jenkins.class.getName()+".slaveAgentPort", slaveAgentPort);
```

Comment:
If the port is not enforced on each start, then people will still using 'hacks' such as https://github.com/jenkinsci/docker/blob/master/init.groovy, which is what I want to avoid.

I would actually go even further: if this property is set, the jnlp port shouldn't be editable through UI. Most cases this port is fixed because of firewall setup so you don't want anyone to change the value through UI once started.